tests/fractal-async-client/test_matrix.py

```diff
@@ -0,0 +1,485 @@
+    client_instance.close = AsyncMock()
+    homeserver_url = "https://matrix-client.test.com"
+    discovery_info_response = DiscoveryInfoResponse(homeserver_url)
+    discovery_info_response.transport_response = MagicMock(spec=TransportResponse)
+    discovery_info_response.transport_response.ok = False
+    discovery_info_response.transport_response.reason = "Error"
+    response = AsyncMock(return_value=discovery_info_response)
+    client_instance.discovery_info = response
+    server, apexchanged = await get_homeserver_for_matrix_id("@user:matrix.org")
+    assert server == homeserver_url
+    assert apexchanged == True
+    client_instance.discovery_info.assert_awaited()
+
+
+async def test_register_with_token_works():
+    homeserver_url = "http://localhost:8008"
+    access_token = "test_token"
+    client = FractalAsyncClient(homeserver_url, access_token)
+    matrix_id = "@user:localhost"
+    password = "pass"
+    registration_token = "test_reg"
+    register_with_token_response = RegisterResponse(matrix_id, "devid", access_token)
+    client.register_with_token = AsyncMock(return_value=register_with_token_response)
+    token = await client.register_with_token(matrix_id, password, registration_token)
+
+
+async def test_get_room_invites_sync_error():
+    client = FractalAsyncClient()
+    with patch.object(
+        client, "sync", new=AsyncMock(return_value=SyncError("Error with request"))
+    ):
+        with pytest.raises(Exception) as e:
+            await client.get_room_invites()
+        assert "Error with request" in str(e.value)
+
+
+async def test_get_room_invites_return_inviteinfo():
+    client = FractalAsyncClient()
+    sample_next_batch = "sample_batch_value"
+    rooms = Rooms(
+        invite={"invite_room_id": InviteInfo(invite_state=[])},
+        join={
+            "join_room_id": RoomInfo(
+                timeline=Timeline(events=[], limited=True, prev_batch=""),
+                state=[],
+                ephemeral=[],
+                account_data=[],
+            )
+        },
+        leave={
+            "leave_room_id": RoomInfo(
+                timeline=Timeline(events=[], limited=True, prev_batch=""),
+                state=[],
+                ephemeral=[],
+                account_data=[],
+            )
+        },
+    )
+    devicelist = DeviceList(changed=[], left=[])
+    devicecount = DeviceOneTimeKeyCount(curve25519=None, signed_curve25519=None)
+    client.sync = AsyncMock(
+        return_value=SyncResponse(
+            next_batch=sample_next_batch,
+            rooms=rooms,
+            device_key_count=devicecount,
+            device_list=devicelist,
+            to_device_events=[],
+            presence_events=[],
+        )
+    )
+    result = await client.get_room_invites()
+    expected_invite_info = {"invite_room_id": InviteInfo(invite_state=[])}
+    assert result == expected_invite_info
+
+
+async def test_join_room_logger():
```

Comment:
This test should be fleshed out a little more. Let's make sure that join is called